pkg/apis/zookeeper/v1beta1/zookeepercluster_types.go

```diff
@@ -157,18 +155,24 @@ func (s *ZookeeperClusterSpec) withDefaults(z *ZookeeperCluster) (changed bool)
 	if s.Pod.withDefaults(z) {
 		changed = true
 	}
-	if s.Persistence == nil {
-		s.Persistence = &Persistence{}
+	if s.Storage == nil || (s.Storage.Persistence == nil && s.Storage.Storagetype != "ephemeral") {
+		s.Storage = &Storage{}
+		s.Storage.Storagetype = "persistence"
+		s.Storage.Persistence = &Persistence{}
 		changed = true
 	}
-	if s.Persistence.withDefaults() {
+	if s.Storage != nil && s.Storage.Storagetype == "ephemeral" && s.Storage.Ephemeral == nil {
```

Comment:
U can rearrange the code in the following manner to ensure u cover all possible cases
```
if s.Storage == nil {
    ...
}

if s.Storage != nil && s.Storage.Storagetype == "ephemeral" {
       if s.Storage.Ephemeral == nil {
           ....
       } else {
           ....
       }
}

if s.Storage != nil && s.Storage.Storagetype != "ephemeral" {
       if s.Storage.Persistence == nil {
           ....
       } else {
           ....
       }
}
```